### ISSUE TITLE
PUT /api/ciphers/move not implemented

### ISSUE THREAD

**Issue by mprasil** [dani-garcia/bitwarden_rs#99]
New vault (#93) calls `PUT` on `/api/ciphers/move` to "Move selected" ciphers. We have `POST` handler [already implemented](https://github.com/dani-garcia/bitwarden_rs/blob/48e69cebab29585684be4bf22282ff1dd4697d24/src/api/core/ciphers.rs#L486) for this, so this should be easy to fix.

**Comment by krankur** [comment]
Hi, I am new to Rust and wanted to start contributing to open source projects. This bug looked like a good beginner task, so I decided to give it a shot. I forked and cloned the project, but on "cargo run" I am getting the following error message:
```
JWT keys don't exist, checking if OpenSSL is available...
OpenSSL detected, creating keys...
Error creating keys, exiting...
```
Can someone please guide me on what can be the issue here?

**Comment by mprasil** [comment]
Hi, I think the program fails to create the keys. Normally they are in the `data` directory, so make sure that whatever user you're running it under can create that folder and/or write to it.

**Comment by krankur** [comment]
So, finally I found out the issue. It was the old openssl version being used. Even though I had `1.0.2l` version installed through brew, the old version at `/usr/local/bin/openssl` was being picked up. I had to create a symlink to the newer version to make it work. https://apple.stackexchange.com/questions/126830/how-to-upgrade-openssl-in-os-x

**Comment by krankur** [comment]
Another question. While doing the setup I was following the instructions in `Build.md` file. Some of the instructions are out of date now, as in the latest vault release, project structure has been changed and the file `settings.Production.json` no longer exists. I had to use the previous release to build the project. Is the build process with the latest vault version documented anywhere?

Also, as per my understanding, the aim of this task is to accommodate the changes introduced as part of the latest vault release. So, to work on this bug I will need to setup the new vault. Is my understanding correct?

**Comment by dani-garcia** [comment]
It’s not documented at the moment, but you can look at the lines 16-25 of the dockerfile to get an idea of the setup.

Basically, clone the 2.0 repo, apply the patch and run the npm commands.

**Comment by mqus** [comment]
I packaged vault for archlinux and as of my knowledge, you don't have to change anything anymore and the build process is as following (taken from build.sh in the vault repo):

	$ npm install
	$ npm run sub:update
	$ npm run dist:selfhost
and the output directory is then `build/`

I have tested it and it worked fairly well but I can still be wrong.

**Comment by mprasil** [comment]
Note that this is still beta branch, so not ready yet. There are couple bugs that need to be fixed (like this one) before the new Vault (2.0) is usable.

**Comment by krankur** [comment]
Thanks to all. I was able to setup with Vault (2.0). Will start working on the bug now.

**Comment by dani-garcia** [comment]
This should be fixed by #108
